edweiss-app/components/lectures/slides/StudentQuestion.tsx

```diff
@@ -1,55 +1,146 @@
 import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
 import TView from '@/components/core/containers/TView';
+import For from '@/components/core/For';
 import Icon from '@/components/core/Icon';
 import TText from '@/components/core/TText';
 import FancyTextInput from '@/components/input/FancyTextInput';
-import { callFunction } from '@/config/firebase';
+import { callFunction, Document } from '@/config/firebase';
+import SyncStorage from '@/config/SyncStorage';
 import ReactComponent from '@/constants/Component';
+import { useAuth } from '@/contexts/auth';
 import LectureDisplay from '@/model/lectures/lectureDoc';
-import { t } from 'i18next';
-import { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { ActivityIndicator } from 'react-native';
+import Toast from 'react-native-toast-message';
 
-interface CustomDocument<T> {
-    data: T;
-}
 
-const StudentQuestion: ReactComponent<{ courseName: string, lectureId: string, questionsDoc: CustomDocument<LectureDisplay.Question>[] | undefined }> = ({ courseName, lectureId, questionsDoc }) => {
+const StudentQuestion: ReactComponent<{ courseName: string, lectureId: string, questionsDoc: Document<LectureDisplay.Question>[] | undefined }> = ({ courseName, lectureId, questionsDoc }) => {
     const [question, setQuestion] = useState<string>('');
     const [isLoading, setIsLoading] = useState<boolean>(false);
+    const [username, setUsername] = useState<string>('');
+    const [isAnonym, setIsAnonym] = useState<boolean>(false);
+    const [enableDisplay, setEnableDisplay] = useState<boolean>(false);
+    const { uid } = useAuth();
 
-    // Display each question given as parameters as a component 
-    const renderQuestion = (question: string, key: React.Key) => (
-        <TView key={key} mb={'sm'} backgroundColor='crust' borderColor='surface0' radius={14} flex={1} flexDirection='column' ml='sm'>
-            <TText ml={16} mb={4} size={'sm'} pl={2} pt={'sm'} color='overlay2'>{t(`showtime:peer_question`)}</TText>
-            <TView pr={'sm'} pl={'md'} pb={'sm'} flexDirection='row' justifyContent='flex-start' alignItems='center'>
-                <TText ml={10} color='overlay0'>{question}</TText>
-            </TView>
-        </TView>
-    );
+
+    useEffect(() => {
+        const initializeStorage = async () => {
+            await SyncStorage.init();
```

Comment:
SyncStorage is already initialized by default and should **not** be initialized again. Is there something I'm missing?